apps/studio/pages/project/[ref]/functions/new.tsx

```diff
@@ -0,0 +1,286 @@
+import { useEffect, useState } from 'react'
+import { CornerDownLeft, Loader2, Book, Check } from 'lucide-react'
+import { Button, Input_Shadcn_, Label_Shadcn_, cn } from 'ui'
+import { AiIconAnimation } from 'ui'
+import { BASE_PATH } from 'lib/constants'
+import { useSelectedProject } from 'hooks/misc/useSelectedProject'
+import { useOrgOptedIntoAi } from 'hooks/misc/useOrgOptedIntoAi'
+import { IS_PLATFORM } from 'lib/constants'
+import { useEdgeFunctionDeployMutation } from 'data/edge-functions/edge-functions-deploy-mutation'
+import { toast } from 'sonner'
+import { useRouter } from 'next/router'
+import { useParams } from 'common'
+import { useAppStateSnapshot } from 'state/app-state'
+import DefaultLayout from 'components/layouts/DefaultLayout'
+import EdgeFunctionsLayout from 'components/layouts/EdgeFunctionsLayout/EdgeFunctionsLayout'
+import { PageLayout } from 'components/layouts/PageLayout/PageLayout'
+import FileExplorerAndEditor from 'components/ui/FileExplorerAndEditor/FileExplorerAndEditor'
+import { EDGE_FUNCTION_TEMPLATES } from 'components/interfaces/Functions/Functions.templates'
+import {
+  Command_Shadcn_,
+  CommandEmpty_Shadcn_,
+  CommandGroup_Shadcn_,
+  CommandInput_Shadcn_,
+  CommandItem_Shadcn_,
+  CommandList_Shadcn_,
+  Popover_Shadcn_,
+  PopoverContent_Shadcn_,
+  PopoverTrigger_Shadcn_,
+} from 'ui'
+import { useFlag } from 'hooks/ui/useFlag'
+
+const NewFunctionPage = () => {
+  const router = useRouter()
+  const { ref } = useParams()
+  const project = useSelectedProject()
+  const isOptedInToAI = useOrgOptedIntoAi()
+  const includeSchemaMetadata = isOptedInToAI || !IS_PLATFORM
+  const { setAiAssistantPanel } = useAppStateSnapshot()
+  const edgeFunctionCreate = useFlag('edgeFunctionCreate')
+
+  // TODO (Saxon): Remove this once the flag is fully launched
+  useEffect(() => {
+    if (!edgeFunctionCreate) {
+      router.push(`/project/${ref}/functions`)
+    }
+  }, [edgeFunctionCreate, ref, router])
+
+  const [files, setFiles] = useState<
+    { id: number; name: string; content: string; selected?: boolean }[]
+  >([
+    {
+      id: 1,
+      name: 'index.ts',
+      selected: true,
+      content: `// Setup type definitions for built-in Supabase Runtime APIs
+import "jsr:@supabase/functions-js/edge-runtime.d.ts";
+
+console.info('server started');
+
+Deno.serve(async (req: Request) => {
+  const data = {
+    message: 'Hello from Supabase Edge Functions!',
+  };
+
+  return new Response(
+    JSON.stringify(data),
+    { headers: { 'Content-Type': 'application/json', 'Connection': 'keep-alive' }}
+  );
+});`,
+    },
```

Comment:
just wondering should we just use the hello world example from Functions.templates.ts here?